### PR TITLE
Stop using the deprecated release_tags api

### LINKED ISSUE #2740: Decouple from the releaseTags API
Use the cocina API instead.

Depends on https://github.com/sul-dlss/dor-services-app/issues/2834 (for writing)

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

This is now available via the cocina api.

Fixes #2740

### CODE REVIEW

File: app/jobs/release_object_job.rb [hunk 1]
```diff
@@ -67,6 +63,34 @@ def create_release_tag(bulk_action, cocina, log)
     false
   end
 
+  # Return a copy of the existing cocina model with a new tag appended
+  def model_with_new_release_tag(cocina)
```
Comment (justinlittman) [inline app/jobs/release_object_job.rb:67]: Should this be using a change set?

Comment (jcoyne) [reply app/jobs/release_object_job.rb:67]: I don't think it needs to, but we could make a new change set object just for this sort of change. I think the indirection between having another class might make it slightly harder to follow than this simpler approach.

Comment (justinlittman) [reply app/jobs/release_object_job.rb:67]: I don't want to delay this ticket, so happy to have this discussion sync. Couldn't it just be added to an existing change set class?

Comment (jcoyne) [reply app/jobs/release_object_job.rb:67]: I'm not keen to have one set of functionality in a change set class that is used by one consumer and then a disjoint set of functionality in the same class used by another consumer.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**